### PR TITLE
SecretServer save multiple workload's key cert.

### PR DESCRIPTION
**Pull request by incfly**
Efforts for #4483 . The end result will be each key/certs stored in its own directory, `/etc/certs/<service_account>/{key/cert.pem}`.

- Stop using SecretServer to save CAClient's own credentials. This should reduce the confusion by having SS only managing workload's key cert, but feel free to propose something else.
- Changing SetKey & SetCerts into single method Save for consistency.
- Remove FileUtil & its mocking, use the implementation to test directly.

Extracting service account from KeyCertBundle is not implemented yet.

@mangchiandjjoe

### CODE REVIEW

File: security/cmd/node_agent/management/server.go [hunk 1]
```diff
@@ -158,12 +160,14 @@ func (s *Server) WorkloadAdded(ctx context.Context, request *pb.WorkloadInfo) (*
 	if err != nil {
 		return nil, fmt.Errorf("csr request failed for service %v %v", sa, err)
 	}
-
-	if err := s.ss.SetServiceIdentityPrivateKey(priv); err != nil {
-		return nil, fmt.Errorf("failed in set private key for service account %v, uid %v err %v", sa, uid, err)
+	// TODO: get root somewhere?
+	kb, err := pkiutil.NewVerifiedKeyCertBundleFromPem(resp.SignedCert, priv, resp.CertChain, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build key cert bundle")
```
Comment (wattli) [inline security/cmd/node_agent/management/server.go]: log 'err' as well?

File: security/cmd/node_agent/management/server.go [hunk 2]
```diff
@@ -158,12 +160,14 @@ func (s *Server) WorkloadAdded(ctx context.Context, request *pb.WorkloadInfo) (*
...
 	}
-	if err := s.ss.SetServiceIdentityCert(resp.SignedCert); err != nil {
-		return nil, fmt.Errorf("failed in set cert for service account %v, uid %v err %v", sa, uid, err)
+
+	if err := s.ss.Save(kb); err != nil {
```
Comment (wattli) [inline security/cmd/node_agent/management/server.go]: I feel ss is not very readable. Expand it?

File: security/cmd/node_agent/na/nodeagent_test.go [hunk 3]
```diff
@@ -166,9 +152,14 @@ func TestStartWithArgs(t *testing.T) {
 			t.Errorf("Test case [%s]: sendCSR is called incorrect times: %d VS (expected) %d",
 				id, c.cAClient.Counter, c.sendTimes)
 		}
-		if c.fileContent != nil && !bytes.Equal(fakeFileUtil.WriteContent["cert_file"], c.fileContent) {
-			t.Errorf("Test case [%s]: cert file content incorrect: %s VS (expected) %s",
-				id, fakeFileUtil.WriteContent["cert_file"], c.fileContent)
-		}
+		// TODO(incfly): add test back after we have Read/Load methods on SecretServer.
```
Comment (wattli) [inline security/cmd/node_agent/na/nodeagent_test.go]: This does not look good, please consider to remove from master branch. You can save it to personal branch if necessary.

File: security/pkg/pki/util/keycertbundle.go [hunk 4]
```diff
@@ -110,6 +110,12 @@ func NewKeyCertBundleWithRootCertFromFile(rootCertFile string) (*KeyCertBundleIm
 	}, nil
 }
 
+// TODO(incfly): temporarily workaround of interface adding methods.
+// TODO: implement this.
```
Comment (wattli) [inline security/pkg/pki/util/keycertbundle.go]: Remove this extra TODO?

File: security/cmd/node_agent/management/server.go (hunk 1 above)
Comment (incfly) [reply security/cmd/node_agent/management/server.go]: Done.

File: security/cmd/node_agent/management/server.go (hunk 2 above)
Comment (incfly) [reply security/cmd/node_agent/management/server.go]: Changed to secretServer

File: security/cmd/node_agent/na/nodeagent_test.go (hunk 3 above)
Comment (incfly) [reply security/cmd/node_agent/na/nodeagent_test.go]: Removed and reword the todo

File: security/pkg/pki/util/keycertbundle.go (hunk 4 above)
Comment (incfly) [reply security/pkg/pki/util/keycertbundle.go]: Done.

**Pull request merged**